### PR TITLE
SDA-3861_20: Upgrade SDA to latest Electron 20 release

### PR DESCRIPTION
**Pull request by NguyenTranHoangSym**
## Description

Pump Electron to 20.0.x

### Before

Put screenshots, videos or gif demos of what was the behaviour before your PR.

### After

Put screenshots, videos or gif demos of what is the behaviour after your PR.

## Quality marks

- [ ] Code quality is according to [guidelines](https://github.com/SymphonyOSF/SFE-Lite/blob/master/docs/coding-guidelines.md)
- [ ] Unit Tests according to [guidelines](https://github.com/SymphonyOSF/SFE-Lite/blob/master/docs/unit-test-guidelines.md)
- [ ] E2E Tests according to [guidelines](https://github.com/SymphonyOSF/SFE-Lite/blob/master/docs/e2e-test-guidelines.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -176,7 +176,7 @@
     "builder-util-runtime": "^9.0.3",
     "cross-env": "5.2.0",
     "del": "3.0.0",
-    "electron": "^19.0.10",
+    "electron": "^20.1.3",
```
Comment (sbenmoussati) [inline package.json:179]: Please change "version" field to 20.0.0 (line 4)

**Comment by NguyenTranHoangSym** [comment]
<html>
<body>


Environment | Changes
-- | --
Mac-only | Added app.isHidden()


<p>*Added an API to check if the app is hidden on macOS. This is a distinct state from the app windows being hidden, and can be used to implement unique behaviour on activation. It is also useful in conjunction with NSWindow.canHide, which I will introduce in a follow-up MR if there are no objections. |
| All | Added browserDatato app.setPath/getPath</p>
<p>*This is very useful for apps that do not store user information in browser, as they treat browser data as cache and want to store them in cache location|</p>
<p>*Before this PR apps can achieve similar effect by changing the path of userData, but we should separate sessionData from userData as apps may want to store user information in userData while storing browser data in cache location |
| Windows-only | Added immersive dark mode on Windows</p>
<p>*A refactored implementation of dark mode for Windows, using official API for Windows 11 and undocumented flag on Win 10 &gt;= 20H1. It allows Electron apps to set a dark title bar based on the system preference. |
| All  | Added support for HIDDevice.forget()</p>
<p>*This PR implements HIDDevice.forget() and also sets up permissioning to allow enabling forget() functionality for Web Serial and Web Bluetooth in future PRs
<a href="https://chromium-review.googlesource.com/c/chromium/src/+/3297868">https://chromium-review.googlesource.com/c/chromium/src/+/3297868</a> |
| Mac-only | Added support for panel-like behavior. Window can float over full-screened apps</p>
<ul>
<li>Created a new ElectronNSPanel window which inherits from ElectronNSWindow. This window mimics the behavior of NSPanel - visible on top of full screened apps, joins all spaces. To achieve this it adds the &quot;NSWindowStyleMaskNonactivatingPanel&quot; style mask, normally reserved for NSPanel, at runtime. |</li>
</ul>

</body>
</html>

File: package.json (hunk 1 above)
Comment (NguyenTranHoangSym) [reply package.json:179]: Should I make it a solid version of 20.0.0 or it will just be ok with the @latest?
@sbenmoussati

Comment (sbenmoussati) [reply package.json:179]: solid version "20.0.0" is ok @NguyenTranHoangSym

**Comment by NguyenTranHoangSym** [comment]
**Few version notes**

**Mac-only**
_Added app.isHidden()_

*Added an API to check if the app is hidden on macOS. This is a distinct state from the app windows being hidden, and can be used to implement unique behaviour on activation. It is also useful in conjunction with NSWindow.canHide, which I will introduce in a follow-up MR if there are no objections.

_Added support for panel-like behavior. Window can float over full-screened apps_

* Created a new ElectronNSPanel window which inherits from ElectronNSWindow. This window mimics the behavior of NSPanel - visible on top of full screened apps, joins all spaces. To achieve this it adds the "NSWindowStyleMaskNonactivatingPanel" style mask, normally reserved for NSPanel, at runtime.

**Windows-only**

_Added immersive dark mode on Windows_

*A refactored implementation of dark mode for Windows, using official API for Windows 11 and undocumented flag on Win 10 >= 20H1. It allows Electron apps to set a dark title bar based on the system preference.

**All**

_Added support for HIDDevice.forget()_

*This PR implements HIDDevice.forget() and also sets up permissioning to allow enabling forget() functionality for Web Serial and Web Bluetooth in future PRs
https://chromium-review.googlesource.com/c/chromium/src/+/3297868

_Added browserDatato app.setPath/getPath_

*This is very useful for apps that do not store user information in browser, as they treat browser data as cache and want to store them in cache location|

*Before this PR apps can achieve similar effect by changing the path of userData, but we should separate sessionData from userData as apps may want to store user information in userData while storing browser data in cache location

**Review by KiranNiranjan — APPROVED** [review]

**Review by sbenmoussati — APPROVED** [review]

**Pull request merged**